packages/frontend-shared/src/store/login-connect-store.ts

```diff
@@ -51,6 +52,7 @@ export const useLoginConnectStore = defineStore({
 
   state (): LoginConnectState {
     return {
+      hasInitiallyLoaded: false,
```

Comment:
this seems like a good addition!

Reply:
I feel like this should be a derived value from other variables using `computed`, not an independently configured flag, though - by having a separate flag we lose the knowledge of what actually constitutes loading as completed. That might be  impractical or a big lift at this point in time, though.

I did a `console.log`, `setHasInitiallyLoaded` gets called a *lot* of times - not really a problem per-se, but the app should only be initially loaded once. It's also kind of hard to know where to look - does the load state of the entire app really belong in the `loginConnect` store? This is associated with user authentication. I don't really know where this *would* be best located, though. Any ideas @marktnoonan? You probably know the UI best at this point.


This fix does have some pros, though - it's simple, just gotta set a boolean. :100:

Reply:
Good points! It gets called a lot because it is happening inside a `watchEffect` in `CloudViewerAndProject.vue`. Since it's a one-way function, I guess it hurts nothing to keep calling it, but it would probably still be easier to follow if we guard it by checking if it was previously set inside the `watchEffect` and maybe adding a comment to clarify.

The name could also be tweaked. I don't read it as really saying "the app has loaded". I read it as saying "data from graphql has been received and the first batch of flags have been set in the `loginConnectStore`" - so very much a `hasInitiallyLoaded` in a local sense related to the store itself. 

This would it explain why it's at the very end of that `watchEffect`. It's currently only guarded by

```js
  if (!query.data.value) {
    return
  }
```

but appears many lines later in the file. I was going to suggest moving it to the top for clarity, right after that clause, but then it would be before the flags are set, which would defeat its purpose. @warrensplayer I'm just inferring this from what I see here, please correct me on this if needed. I assume what this is avoiding is kind of a ripple through multiple states as each flag is set, before the others.

We _could_ derive this in the store by looking at the values that are changing, but I think the current setup is OK, and has the benefit of being simple. Maybe `haveFirstFlagsBeenSet` or something is a better name for this. 

I feel like pinia can/should batch these updates from this `watchEffect`, maybe we missed the way to do that and could come back to this in the future - to sort of answer what you raised [here](https://github.com/cypress-io/cypress/pull/25539/files?diff=unified&w=1#r1088532619).